### PR TITLE
Gather artifacts in createorupdate

### PR DESCRIPTION
**Pull request by kargakis**
/cc mjudeikis charlesakalugwu

### CODE REVIEW

File: cmd/createorupdate/createorupdate.go [hunk 1]
```diff
@@ -441,6 +449,146 @@ func createResourceGroup(log *logrus.Entry) error {
...
+		return err
+	}
+
+	// gather control plane logs
+	ns := "kube-system"
+	if err := gatherLogs(kc, artifactDir, ns, "sync-master-000000"); err != nil {
+		return err
+	}
+	// TODO: Get logs from the api server and etcd dynamically by using the master count
+	if err := gatherLogs(kc, artifactDir, ns, "master-etcd-master-000000"); err != nil {
```
Comment (mjudeikis) [inline cmd/createorupdate/createorupdate.go]: I think at some point we need to move pod names to constants, as they starting to appear all around the code

File: cmd/createorupdate/createorupdate.go [hunk 2]
```diff
@@ -441,6 +449,146 @@ func createResourceGroup(log *logrus.Entry) error {
 	return err
 }
 
+func gatherArtifacts(artifactDir, artifactKubeconfig string) error {
```
Comment (mjudeikis) [inline cmd/createorupdate/createorupdate.go]: Does it make sense to move these to a separate file as createorupdate.go is becoming kinda big?

File: cmd/createorupdate/createorupdate.go (hunk 1 above)
Comment (kargakis) [reply cmd/createorupdate/createorupdate.go]: There is a TODO just above about constructing these names dynamically.

File: cmd/createorupdate/createorupdate.go (hunk 2 above)
Comment (kargakis) [reply cmd/createorupdate/createorupdate.go]: We will have to add the new file in the `go run` invocation, and I have been resisting it so far but I definitely agree this file should be split into separate self-contained files. Certainly the server code deserves its own `server.go` place.

Comment (kargakis) [reply cmd/createorupdate/createorupdate.go]: Will do the split in a follow-up

**Pull request merged**